Sources/NIOPostgres/Data/PostgresRow.swift

```diff
@@ -2,50 +2,64 @@ public struct PostgresRow: CustomStringConvertible {
     final class LookupTable {
         let rowDescription: PostgresMessage.RowDescription
         let resultFormat: [PostgresFormatCode]
-        
+
+        struct Entry {
+            let indexInRow: Int
+            let field: PostgresMessage.RowDescription.Field
+        }
+        let columnNameToIndexLookupTable: [String: [Entry]]
+
         init(
             rowDescription: PostgresMessage.RowDescription,
             resultFormat: [PostgresFormatCode]
-        ) {
+            ) {
             self.rowDescription = rowDescription
             self.resultFormat = resultFormat
+
+            var columnNameToIndexLookupTable: [String: [Entry]] = [:]
+            for (fieldIndex, field) in rowDescription.fields.enumerated() {
+                columnNameToIndexLookupTable[field.name, default: []].append(.init(indexInRow: fieldIndex, field: field))
+            }
+            self.columnNameToIndexLookupTable = columnNameToIndexLookupTable
```

Comment:
Should we populate this lookup table lazily? We could make the lookup table optional, then populate it on the first call to `lookup` when it is `nil`. That way we don't add any extra computation where the results just end up being discarded anyway.

Reply:
Done.